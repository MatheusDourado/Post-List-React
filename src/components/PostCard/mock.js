//mock = dummy data === 'Dados Ficticios'
export const postCardPropsMock = {
	title: "title PostCard 1",
	body: "body PostCard 1",
	id: 1,
	cover: "img/img.png",
};
