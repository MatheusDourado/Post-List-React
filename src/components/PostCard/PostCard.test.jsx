import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { postCardPropsMock } from "./mock";

const props = postCardPropsMock;

describe("<PostCard />", () => {
	test("should render PostCard correctly", () => {
		render(<PostCard {...props} />);

		expect(screen.getByAltText(props.title)).toHaveAttribute(
			"src",
			props.cover
		);
		expect(
			screen.getByRole("heading", { name: props.title + " 1" })
		).toBeInTheDocument();
		expect(screen.getByText(props.body)).toBeInTheDocument();
	});

	test("should match snaphot", () => {
		const { container } = render(<PostCard {...props} />);

		expect(container.firstChild).toMatchSnapshot();
	});
});
