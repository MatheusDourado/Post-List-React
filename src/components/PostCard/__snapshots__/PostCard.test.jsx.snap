// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snaphot 1`] = `
<div
  class="post"
>
  <img
    alt="title PostCard 1"
    class="post-img"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2
      class="title-1"
    >
      title PostCard 1
       
       1
    </h2>
    <p
      class="text"
    >
      body PostCard 1
    </p>
  </div>
</div>
`;
